.github/CONTRIBUTING.md

```diff
@@ -212,21 +212,43 @@ cmake --build build
 
 ### Explanation of the SDist/wheel building design
 
-> These details are _only_ for building from the Python sources from git. The
+> These details below are _only_ for packaging the Python sources from git. The
 > SDists and wheels created do not have any extra requirements at all and are
 > completely normal.
 
-In order to support CMake output files (`pybind11Config.cmake` and helper
-files), the `[globals]` option, and a simple version file taken from the C++
-sources, the `setup.py` _in the source_ has a few tricks to make the simplest
-possible SDists and wheels. If you want to build the SDist from the GitHub
-source, it is best to have Pip 10 on a manylinux1, macOS, or Windows system.
-You can then build the SDists, or run any procedure that makes SDists
-internally, like making wheels or installing. Since Pip itself
-does not have an `sdist` command (it does have `wheel` and `install`), you will
-need to use the upcoming `build` package:
+The main objective of the packaging system is to create SDists (Python's source
+distribution packages) and wheels (Python's binary distribution packages) that
+include everything that is needed to work with pybind11, and which can be
+installed without any additional dependencies. This is more complex than it
+appears: in order to support CMake as a first class language even when using
+the PyPI package, they must include the _generated_ CMake files (so as not to
+require CMake when installing the `pybind11` package itself). They should also
+provide the option to install to the "standard" location
+(`<ENVROOT>/include/pybind11` and `<ENVROOT>/share/cmake/pybind11`) so they are
+easy to find with CMake, but this can cause problems if you are not an
+environment or using ``pyproject.toml`` requirements. This was solved by having
+two packages; the "nice" pybind11 package that stores the includes and CMake
+files inside the package, that you get access to via functions in the package,
+and a `pybind11-global` package that can be included via `pybind11[global]` if
+you want the more invasive but discoverable file locations.
```

Comment:
this intro paragraph is so much better now, thanks for that!

Reply:
One more stupid question: what does `pip install pybind11[global]` do if `pybind11` is already installed?

Reply:
It just installs `pybind11-global`. Just as if you wrote `pip install pybind11 pybind11-global` and already have pybind11 installed. The only special feature is that they are version-locked.

Reply:
Great, thank you for the clarification.